Use burner-core for token balances

dmihal opened the pull request:
PR #211 integrated burner-core for reading the main 3 assets (xDai, Dai and Eth), this PR expands on that by using burner-core to read the balance of an additional token.

The largest impact on the codebase is the removal of the ERC20TOKEN and ERC20VENDOR variables, replacing them with data from assets.js, as well as hardcoded contract names.

dmihal commented:
Updated this to use the newest packages, and rebased to the recent changes with burner-core

TimDaub commented on src/App.js:
```diff
@@ -1200,11 +1099,10 @@ render() {
                   />
                 </div>
               )
-            }else if(this.state.isVendor&&this.state.isVendor.isAllowed){
```
Why can we remove `this.state.isVendor.isAllowed`?

dmihal replied:
Good catch, we can't :)

I was originally working on abstracting the VendingMachine contract, but it got a little complex and out of scope. I've reverted this line

TimDaub commented on src/App.js:
```diff
@@ -1748,18 +1646,18 @@ render() {
                     buttonStyle={buttonStyle}
                     changeAlert={this.changeAlert}
-                    sendWithLink={(amount,cb)=>{
+                    sendWithLink={async (amount, cb)=>{
```
It looks like we're not taking advantage of labeling that function `async`, so I also think we shouldn't declare it as such.

dmihal replied:
True, reverted

TimDaub commented on src/assets.js:
```diff
@@ -0,0 +1,31 @@
+export let WEB3_PROVIDER = 'https://dai.poa.network';
+export let ERC20IMAGE;
+export let LOADERIMAGE = burnerlogo;
+export let XDAI_PROVIDER = 'https://dai.poa.network'
```
Just an FYI: In the LeapDAO repo we've gotten rid of `WEB3_PROVIDER` and `XDAI_PROVIDER`. Here they're the same value, so technically they should be a single variable too.

dmihal replied:
Eventually all the provider code will be removed, as it is already handled by burner-core, so I'm not sure if it makes sense to integrate that part of the config file here.

TimDaub replied:
OK

TimDaub commented on src/components/Exchange.js:
```diff
@@ -272,9 +273,14 @@ export default class Exchange extends React.Component {
 
 
 
+<<<<<<< HEAD
```
Ups. Please remove :)

dmihal replied:
Ah not sure how i missed this 😱

TimDaub commented on src/components/Exchange.js:
```diff
@@ -1087,29 +1093,26 @@ export default class Exchange extends React.Component {
-      if(this.props.contracts){
-        link = "https://blockscout.com/poa/dai/address/"+this.props.contracts[this.props.ERC20TOKEN]._address+"/contracts"
-      }
+      const link = `https://blockscout.com/poa/dai/address/${token.address}/contracts`;
```
More LeapDAO/burner-wallet config.js propaganda: https://github.com/leapdao/burner-wallet/blob/master/src/components/Receive.js#L45 :)

dmihal replied:
This part does seem like it would make sense in a config file, since it's only client-facing. However, it's probably out of scope for this PR :)

TimDaub replied:
Yes it is out of scope. Just wanted to promote config.js :D

TimDaub commented on src/core.js:
```diff
@@ -23,6 +24,9 @@ if (process.env.REACT_APP_MODE === 'local') {
 
 const assets = [mainAsset, dai, eth];
+if (token) {
+  assets.unshift(token);
```
Mind elaborating what this does?

dmihal replied:
Assets is an array of all tokens/coins we will display in the app. If the token variable is set here, then we want to add that to the array so we can display it in the list of tokens.

TimDaub replied:
OK, I believe a comment would clear that up nicely. But up to you.

TimDaub replied:
Actually let's merge it and comment it later.

dmihal commented:
Thanks for the review @TimDaub, you caught some good issues! I've addressed all the issues.

This PR hasn't actually been tested on a token in a bit, I'll try to do a local test soon.

TimDaub commented:
> This PR hasn't actually been tested on a token in a bit, I'll try to do a local test soon.

OK let me know how it goes and feel free to merge after testing! Thx
